### PR TITLE
add segment 3

### PR DESCRIPTION
**Pull request by mfdeveloper508**

### CODE REVIEW

File: segments/base_segment_test.go [hunk 1]
```diff
@@ -38,7 +29,7 @@ func (s *SegmentTest) TestBaseSegmentWithIdentificationNumber(c *check.C) {
 	segment.IdentificationNumber = ""
 	err = segment.Validate()
 	c.Assert(err, check.Not(check.IsNil))
-	c.Assert(err, check.DeepEquals, ErrRequired)
+	c.Assert(err, check.DeepEquals, ach.ErrFieldRequired)
```
Comment (wadearnold) [inline segments/base_segment_test.go:32]: Remove the dependency on the ACH project and copy over the code to Metro2

File: segments/base_segment_test.go [hunk 2]
```diff
@@ -154,7 +105,7 @@ func (s *SegmentTest) TestPackedBaseSegmentWithIdentificationNumber(c *check.C)
...
```
Comment (wadearnold) [inline segments/base_segment_test.go:108]: Remove the dependency on the ACH project and copy over the code to Metro2

File: segments/header_record.go [hunk 3]
```diff
@@ -163,19 +170,20 @@ func (s *HeaderRecord) String() string {
...
-			return ErrSegmentParse
+			return ErrValidField
 		}
 
 		if spec, ok := headerRecordCharacterFormat[fieldName]; ok {
 			if spec.Required == required {
 				fieldValue := fields.FieldByName(fieldName)
 				if fieldValue.IsZero() {
-					return ErrRequired
+					return ach.ErrFieldRequired
```
Comment (wadearnold) [inline segments/header_record.go:186]: Remove the dependency on the ACH project and copy over the code to Metro2

File: segments/header_record.go [hunk 4]
```diff
@@ -266,19 +277,20 @@ func (s *PackedHeaderRecord) String() string {
...
-			return ErrSegmentParse
+			return ErrValidField
 		}
 
 		if spec, ok := headerRecordPackedFormat[fieldName]; ok {
 			if spec.Required == required {
 				fieldValue := fields.FieldByName(fieldName)
 				if fieldValue.IsZero() {
-					return ErrRequired
+					return ach.ErrFieldRequired
```
Comment (wadearnold) [inline segments/header_record.go:293]: Remove the dependency on the ACH project and copy over the code to Metro2

File: segments/header_record_test.go [hunk 5]
```diff
@@ -26,7 +29,7 @@ func (s *SegmentTest) TestTrailerRecordWithInvalidActivityDate(c *check.C) {
 	segment.ActivityDate = 0
 	err = segment.Validate()
 	c.Assert(err, check.Not(check.IsNil))
-	c.Assert(err, check.DeepEquals, ErrRequired)
+	c.Assert(err, check.DeepEquals, ach.ErrFieldRequired)
```
Comment (wadearnold) [inline segments/header_record_test.go:32]: Remove the dependency on the ACH project and copy over the code to Metro2

File: segments/header_record_test.go [hunk 6]
```diff
@@ -53,5 +56,5 @@ func (s *SegmentTest) TestPackedTrailerRecordWithInvalidActivityDate(c *check.C)
...
```
Comment (wadearnold) [inline segments/header_record_test.go:59]: Remove the dependency on the ACH project and copy over the code to Metro2

File: segments/validators.go [hunk 7]
```diff
@@ -5,34 +5,33 @@
...
-	dateRegex              = regexp.MustCompile(`(0[1-9]|1[0-2])(0[1-9]|[1-2][0-9]|3[0-1])[0-9]{4}`)                                       //  MMDDYYYY
 )
 
 type validator struct{}
 
 func (v *validator) isUpperalphanumeric(s string) error {
-	if upperalphanumericRegex.MatchString(s) {
-		return ErrUpperAlpha
+	if upperAlphanumericRegex.MatchString(s) {
+		return ach.ErrUpperAlpha
```
Comment (wadearnold) [inline segments/validators.go:27]: Remove the dependency on the ACH project and copy over the code to Metro2

File: segments/validators.go [hunk 8]
```diff
@@ -5,34 +5,33 @@
...
 	}
 	return nil
 }
 
 func (v *validator) isAlphanumeric(s string) error {
 	if alphanumericRegex.MatchString(s) {
-		return ErrAlphanumeric
+		return ach.ErrNonAlphanumeric
```
Comment (wadearnold) [inline segments/validators.go:34]: Remove the dependency on the ACH project and copy over the code to Metro2

File: segments/validators.go [hunk 9]
```diff
@@ -52,29 +51,16 @@ func (v *validator) isPhoneNumber(number int64) error {
...
-	return "", errors.New("not filled")
-}
-
 func (v *validator) isValidType(elm field, data string) error {
 	if elm.Required == required {
 		if elm.Type&numeric > 0 {
 			val, _ := strconv.Atoi(data)
 			if val == 0 {
-				return ErrRequired
+				return ach.ErrFieldRequired
```
Comment (wadearnold) [inline segments/validators.go:59]: Remove the dependency on the ACH project and copy over the code to Metro2

File: segments/validators.go [hunk 10]
```diff
@@ -52,29 +51,16 @@ func (v *validator) isPhoneNumber(number int64) error {
...
 			}
-		} else if elm.Type&alphanumeric > 0 || elm.Type&alpha > 0 || elm.Type&binaryDescriptor > 0 {
+		} else if elm.Type&alphanumeric > 0 || elm.Type&alpha > 0 || elm.Type&descriptor > 0 {
 			if len(data) == 0 {
-				return ErrRequired
+				return ach.ErrFieldRequired
```
Comment (wadearnold) [inline segments/validators.go:63]: Remove the dependency on the ACH project and copy over the code to Metro2

Comment (mfdeveloper508) [reply segments/validators.go]: fixed

**Pull request merged**